### PR TITLE
feat: `verify email` form and modal migrated to Mantine

### PR DESCRIPTION
**Pull request by yasmin-e**
Closes: #4909 

### Description:
migrated
```
- verify email form (common component)
- verify email page (part of register flow)
- verify email modal (visible in settings > profile > click here to verify email)
- verification successful modal (visible when verification successful in register flow)
```
notes
```
- using Mantine modal for the first time, very simple very little change from Blueprint
- the only thing that's sitll Blueprint is the animated success Icon, tried migrating it, not worth it for now
- renamed CreateUserForm folder RegisterForms, inside there's CreateUserForm and VerifyEmailForm
```
### verify email in register flow

https://user-images.githubusercontent.com/67699259/227534697-6e16823e-35bb-46d3-abe0-73bcf58c0f36.mov

Screenshot 2023-03-24 at 14 18 35
Screenshot 2023-03-24 at 14 18 55
Screenshot 2023-03-24 at 14 19 01
Screenshot 2023-03-24 at 14 19 14
Screenshot 2023-03-24 at 14 19 29

### verify email in settings > profile

https://user-images.githubusercontent.com/67699259/227534436-4a4cb125-c922-4bb5-aab9-a0c0cb16bb01.mov

### CODE REVIEW

File: packages/frontend/src/pages/VerifyEmail.tsx [hunk 1]
```diff
@@ -74,41 +57,37 @@ export const VerifyEmailPage: FC = () => {
...
-                            mutate(code);
-                        }}
-                        onResend={sendVerificationEmail}
-                        isLoading={statusLoading || emailLoading}
-                        verificationLoading={verificationLoading}
-                    />
-                </CardWrapper>
-                <FormFooterCopy>
+            {/* FIXME: use Mantine sizes for width */}
+            <Stack w={400} mt="xl" pt="lg">
```
Comment (IrakliJani) [inline packages/frontend/src/pages/VerifyEmail.tsx:61]: you can rebase on main and use something from this list:
```
        one: rem(1),
        two: rem(2),
        xxs: rem(4),
        xs: rem(8),
        sm: rem(12),
        md: rem(16),
        lg: rem(20),
        xl: rem(24),
        xxl: rem(32),
        '3xl': rem(40),
        '4xl': rem(48),
        '5xl': rem(64),
        '6xl': rem(80),
        '7xl': rem(96),
        '8xl': rem(128),
        '9xl': rem(160),
```

File: packages/frontend/src/pages/VerifyEmail.tsx [hunk 2]
```diff
@@ -74,41 +57,37 @@ export const VerifyEmailPage: FC = () => {
...
+                <Image
+                    src={LightdashLogo}
+                    alt="lightdash logo"
+                    width={130}
```
Comment (IrakliJani) [inline packages/frontend/src/pages/VerifyEmail.tsx:65]: `8xl` I guess? (128px)

File: packages/frontend/src/pages/VerifyEmail.tsx [hunk 3]
```diff
@@ -1,67 +1,50 @@
...
-                <SaveButton
-                    intent={Intent.PRIMARY}
-                    text="Continue"
-                    onClick={onContinue}
-                />
-            </EmailVerifiedWrapper>
-        </EmailVerifiedModal>
+                <Button onClick={onContinue}>Continue</Button>
+            </Stack>
+        </Modal>
```
Comment (IrakliJani) [inline packages/frontend/src/pages/VerifyEmail.tsx:41]: great 🎉

File: packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                        if (completed || data?.otp?.isMaxAttempts) {
+                            return <></>;
+                        }
+                        return (
+                            // FIXME: use Mantine sizes for width
+                            <Stack spacing="xs" mt="md" w={290} align="center">
+                                <Button
+                                    loading={verificationLoading}
+                                    type="submit"
+                                    w={230}
```
Comment (IrakliJani) [inline packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx:102]: if you have padding to the parent Stack element and here you can use w="100%"

File: packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        }
+    }, [data, form.setFieldError, form.clearFieldError]);
+
+    if (loadingState) {
+        return <LoadingState title="" />;
+    }
+
+    return (
+        // FIXME: use Mantine sizes for width
+        <Stack spacing="md" justify="center" align="center" w={300} mx="auto">
```
Comment (IrakliJani) [inline packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx:63]: with the current scale I think this should be 320
```
        one: rem(1),
        two: rem(2),
        xxs: rem(4),
        xs: rem(8),
        sm: rem(12),
        md: rem(16),
        lg: rem(20),
        xl: rem(24),
        xxl: rem(32),
        '3xl': rem(40),
        '4xl': rem(48),
        '5xl': rem(64),
        '6xl': rem(80),
        '7xl': rem(96),
        '8xl': rem(128),
        '9xl': rem(160),
```

we might need to make a scalable utility for size scales (in extreme cases like this)

File: packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+                <Countdown
+                    key={expirationTime?.toString()}
+                    date={expirationTime}
+                    renderer={({ minutes, seconds, completed }) => {
+                        if (completed || data?.otp?.isMaxAttempts) {
+                            return <></>;
+                        }
+                        return (
+                            // FIXME: use Mantine sizes for width
+                            <Stack spacing="xs" mt="md" w={290} align="center">
```
Comment (IrakliJani) [inline packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx:98]: you can remove w=290 if you add a proper padding to the first Stack

**Review by IrakliJani — CHANGES_REQUESTED** [review]
minor changes needed after bumping mantine

File: packages/frontend/src/pages/VerifyEmail.tsx (hunk 1 above)
Comment (yasmin-e) [reply packages/frontend/src/pages/VerifyEmail.tsx]: done 👍🏼

File: packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx (hunk 4 above)
Comment (yasmin-e) [reply packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx]: done 👍🏼

File: packages/frontend/src/pages/VerifyEmail.tsx (hunk 2 above)
Comment (yasmin-e) [reply packages/frontend/src/pages/VerifyEmail.tsx:65]: leaving hardcoded widths for now, added comments to find them easily!

File: packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx (hunk 5 above)
Comment (yasmin-e) [reply packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx:63]: leaving hardcoded widths for now, added comments to find them easily!

File: packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx (hunk 6 above)
Comment (yasmin-e) [reply packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx]: this is needed because of width fluctuation with the countdown

**Review by IrakliJani — APPROVED** [review]
this flow is so good!
approved, just one comment about a comment without a comment literal 😬

**Pull request merged**